test/unit/calculators/self_assessment_penalties_test.rb

```diff
@@ -316,7 +316,7 @@ def setup
           assert_equal 6132, @calculator.total_owed
           @calculator.payment_date = Date.parse("2025-08-03")
           assert_equal 750, @calculator.late_payment_penalty
-          assert_equal 6306, @calculator.total_owed
+          assert_equal 6330, @calculator.total_owed
```

Comment:
@baisa I'm not entirely sure I understand the test coverage here, I expected to set the date at the new start date of 6th April and anything before would be the old rate, but I didn't see that - I think there is some complicated way it's calculated that I'm not following?

Reply:
@Harriethw would modifing tests like this make it more clear?
```
         @calculator.payment_date = Date.parse("2025-03-03")
          assert_equal 750, @calculator.late_payment_penalty
          assert_equal 6159, @calculator.total_owed
          @calculator.payment_date = Date.parse("2025-05-03")
          assert_equal 750, @calculator.late_payment_penalty
          assert_equal 6223, @calculator.total_owed
```

Reply:
Thanks Barbara - I can't say I understand those dates any better, but I'll take your word for it :) I've made that change